### PR TITLE
Allow `tan` to be used in `df.eval`.

### LINKED ISSUE #55091: BUG: Unsupported operations
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [x] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import numpy as np

matrix = np.random.randint(1, 3, size=(100, 5))
c = ["a", "b", "c", "d", "e"]
df = pd.DataFrame(matrix, columns=c)
transformed_df = df.eval("b_atan = arctan(b)")
transformed_df = transformed_df.drop(["b"], axis=1)
restored_df = transformed_df.eval("b = tan(b_atan)")
```

### Issue Description

While running the example, I have getting the following exception:

ValueError: "tan" is not a supported function

Why "tan" is not supported if "arctan" is supported?
Where I can check the list of all supported mathematical functions?

### Expected Behavior

"tan" function will be supported

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : ca60aab7340d9989d9428e11a51467658190bb6b
python           : 3.9.16.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.22621
machine          : AMD64
processor        : Intel64 Family 6 Model 183 Stepping 1, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : Russian_Ukraine.1251

pandas           : 1.4.4
numpy            : 1.23.5
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.8.0
pip              : 23.1.2
Cython           : None
pytest           : 7.1.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : 1.0.3
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.5.0
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : 2.1.3
matplotlib       : 3.5.3
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.9.3
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None

</details>

### PR DESCRIPTION
**Pull request by domsmrz**
- [x] closes #55091 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature (not explicitly adding brand new tests but there are tests covering all the `_unary_math_ops`.
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] ~Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.~
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by yuanx749** [comment]
While at it, is it feasible to also add other functions? I noticed the functions supported by NumExpr [here](https://numexpr.readthedocs.io/en/latest/user_guide.html#supported-functions), such as conj, real, imag.

**Comment by domsmrz** [comment]
> While at it, is it feasible to also add other functions? I noticed the functions supported by NumExpr [here](https://numexpr.readthedocs.io/en/latest/user_guide.html#supported-functions), such as conj, real, imag.

Probably, but I'd prefer to keep this PR as is, since this is pure straight-forward fix.
Adding `where` is discussed here: #34834 .
I can imagine that adding complex-number operation should be easier than that but I'm not sure how well pandas handles them, so I'd prefer to delegate it to separate PR (at very least it would require adding tests)

File: pandas/core/computation/ops.py [hunk 1]
```diff
@@ -45,6 +45,7 @@
 _unary_math_ops = (
     "sin",
     "cos",
+    "tan",
```
Comment (mroeschke) [inline pandas/core/computation/ops.py:48]: Can you add a test for this?

File: doc/source/whatsnew/v3.0.0.rst [hunk 2]
```diff
@@ -455,6 +455,7 @@ Other
 - Bug in :class:`DataFrame` when passing a ``dict`` with a NA scalar and ``columns`` that would always return ``np.nan`` (:issue:`57205`)
 - Bug in :func:`unique` on :class:`Index` not always returning :class:`Index` (:issue:`57043`)
 - Bug in :meth:`DataFrame.eval` and :meth:`DataFrame.query` which caused an exception when using NumPy attributes via ``@`` notation, e.g., ``df.eval("@np.floor(a)")``. (:issue:`58041`)
+- Bug in :meth:`DataFrame.eval` and :meth:`DataFrame.query` which did not allow to use ``tan`` function
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:458]: Can you add the issue number like the other entries?

File: pandas/core/computation/ops.py (hunk 1 above)
Comment (domsmrz) [reply pandas/core/computation/ops.py:48]: Just by adding "tan" to the list `_unary_math_ops` a standard test (which is applied to all these functions) is automatically added: https://github.com/pandas-dev/pandas/blob/454e2e1d9d7b118953ecfb4edc6f9fe7f5cb07b8/pandas/tests/computation/test_eval.py#L1615
Let me know if you meant anything more specific.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @domsmrz

**Pull request merged**